.azure_pipelines/job_templates/build-verify-publish-template-win.yml

```diff
@@ -0,0 +1,144 @@
+          displayName: Run all unit tests
+
+      # Run ComponentGovernance
+      - task: ComponentGovernanceComponentDetection@0
+        inputs:
+          scanType: 'Register'
+          verbosity: 'Verbose'
+          alertWarningLevel: 'High'
+
+    # Build PyTorch and produce artifacts for verification stage
+    - ${{ if eq(parameters.is_official_build, 'True') }}:
+      # Build PyTorch from source in install mode with Ninja and exclude test binaries
+      - script: call activate $(configuration) && python setup.py install
+        displayName: Build PyTorch from source without test binaries
+
+      # Package PyTorch Wheel
+      - script: call activate $(configuration) && python setup.py bdist_wheel
+        displayName: Package PyTorch Wheel
+
+      # Publish PyTorch Wheel
+      - task: PublishPipelineArtifact@1
+        inputs:
+          targetPath: $(Build.SourcesDirectory)\dist\
+          artifactName: Build_$(Build.BuildNumber)_$(configuration)
+        displayName: Publish PyTorch Wheel to Pipeline Artifacts
+
+  # Verification Stage
+  - ${{ if eq(parameters.verify_stage, 'True') }}:
+    # Download PyTorch Wheel
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifact: Build_$(Build.BuildNumber)_$(configuration)
+        path: $(Build.SourcesDirectory)\verify
+      displayName: Download PyTorch Wheel
+
+    # Install PyTorch Wheel on Windows
+    - script: |
+        call activate $(configuration)
+        cd $(Build.SourcesDirectory)\verify
+        dir torch*win*.whl /b > whl.txt
+        set /p whl= < whl.txt
+        python -m pip install %whl%
+      displayName: Install PyTorch Wheel
+
+    # Ensure PyTorch installed correctly from produced wheel
+    - script: |
+        call activate $(configuration)
+        cd $(Build.SourcesDirectory)\verify
+        python -c "import torch; print('Installed Torch version: ' + torch.__version__)"
+      displayName: Check PyTorch correctly installed from wheel
+
+  # Publishing stage
+  - ${{ if eq(parameters.publish_stage, 'True') }}:
+    # Download PyTorch Wheel
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifact: Build_$(Build.BuildNumber)_$(configuration)
+        path: $(Build.SourcesDirectory)\publish
+      displayName: Download PyTorch Wheel
+
+    # Set up Azure Artifacts for Windows
+    # The pip install --upgrade command is a bug fix for Azure CLI on Windows
+    # More info: https://github.com/Azure/azure-cli/issues/16858
+    - script: |
+        pip install --upgrade pip --target \opt\az\lib\python3.6\site-packages\
+        az extension add -n azure-devops
+      displayName: Set up Azure Artifacts download on Windows
+
+    # Publish wheel to Azure Artifacts
+    - script: |
+        set /p TORCH_VERSION= < version.txt
+        cd $(Build.SourcesDirectory)\publish
+        set YYYY=%DATE:~10,4%
+        set MM=%DATE:~4,2%
+        set DD=%DATE:~7,2%
+        set BUILD_DATE=%YYYY%%MM%%DD%
```

Comment:
This makes artifacts ultimately non-reproducible atfter 24 hours (as BUILD_DATE would be different)
Isn't commit date more important here?

Reply:
That is true, the commit date and hash are the main differentiators. Even the commit date isn't very necessary, however it is good to keep the commit date as it quickly gives an idea on when and which commit was done last. I will change this to use the commit date instead of the build date.